### PR TITLE
enhancement #617: added filtration for event endpoints

### LINKED ISSUE #617: Add event type filtration to all get events endpoints (exсept tg)
You should add filtration by event type query to **getGeneralEvents** and **getGroupEvents**. Ignore tg endpoints!
Don`t change **getGeneralGroupEvents** service method. Create a _wrapper_ around it. 

I should mention, that **getGeneralEvents** and **getGroupEvents** have different query:
- **getGeneralEvents** supports only discipline types, like Lecture, Pratice and Laboratory;
- **getGroupEvents** supports also selective disciplines and "others events", look here:
image

Filtration query example: **getAll** in TeacherController.

### PR DESCRIPTION
**Pull request by SharpDevOps10**
enhancement #617: added filtration for event endpoints

- added disciplineType filtration for general events
- added otherEvents filtration for group events
- updated swagger documentation

closes #617

### CODE REVIEW

**Review by f-ogoog — APPROVED** [review]

File: src/v2/api/services/ScheduleService.ts [hunk 1]
```diff
@@ -67,6 +67,7 @@ export class ScheduleService {
     }
   }
 
+
```
Comment (ArtemLyah) [inline src/v2/api/services/ScheduleService.ts:70]: delete there space

**Review by ArtemLyah — APPROVED** [review]

File: src/v2/api/controllers/ScheduleController.ts [hunk 2]
```diff
@@ -100,11 +115,21 @@ export class ScheduleController {
   async getGeneralEvents (
     @Param('groupId', GroupByIdPipe) id: string,
     @Query('week') week: number,
+    @Query('addLecture') addLecture: boolean,
```
Comment (Hahhhayam) [inline src/v2/api/controllers/ScheduleController.ts:118]: wrap Query in DTO,  like in TeacherController/getAll

File: src/v2/api/controllers/ScheduleController.ts [hunk 3]
```diff
@@ -255,22 +280,57 @@ export class ScheduleController {
...
-      @Request() req,
-      @Param('groupId', GroupByIdPipe) groupId: string,
-      @Query('showOwnSelective', ConvertToBooleanPipe) showOwnSelective,
-      @Query('week') week: number,
+    @Request() req,
+    @Param('groupId', GroupByIdPipe) groupId: string,
+    @Query('showOwnSelective', ConvertToBooleanPipe) showOwnSelective,
+    @Query('week') week: number,
+    @Query('otherEvents') otherEvents: boolean,
+    @Query('addLecture') addLecture: boolean,
```
Comment (Hahhhayam) [inline src/v2/api/controllers/ScheduleController.ts:316]: wrap Query in DTO, like in TeacherController/getAll

**Review by Hahhhayam — CHANGES_REQUESTED** [review]

File: src/v2/api/controllers/ScheduleController.ts [hunk 4]
```diff
@@ -255,22 +277,49 @@ export class ScheduleController {
...
+    name: 'addLecture',
+    type: Boolean,
+    required: false,
+  })
+  @ApiQuery({
+    name: 'addPractice',
+    type: Boolean,
+    required: false,
+  })
+  @ApiQuery({
```
Comment (Hahhhayam) [inline src/v2/api/controllers/ScheduleController.ts:302]: You already described it in DTO, you needn`t it in controller

**Review by Hahhhayam — APPROVED** [review]

**Pull request merged**